### PR TITLE
add ut of cache/v2/listener.go

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
add ut of cache/v2/listener.go
**Which issue(s) this PR fixes**:
a part of #104

### CODE REVIEW

File: pkg/cache/v2/listener_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	core_v2 "kmesh.net/kmesh/api/v2/core"
+	listener_v2 "kmesh.net/kmesh/api/v2/listener"
+	maps_v2 "kmesh.net/kmesh/pkg/cache/v2/maps"
+	"kmesh.net/kmesh/pkg/utils/hash"
+)
+
+func TestListenerFlush(t *testing.T) {
```
Comment (hzxuzhonghu) [inline pkg/cache/v2/listener_test.go:32]: I am thinking you are testing `handleLdsResponse`, is that covered by those tests

Comment (LiZhenCheng9527) [reply pkg/cache/v2/listener_test.go:32]: Yes, I'm testing if the listenercache is handled correctly when maps_v2.ListenerUpdate returns nil.
This covers the handleLdsResponse usage scenario.

File: pkg/cache/v2/listener_test.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2023 The Kmesh Authors.
```
Comment (hzxuzhonghu) [inline pkg/cache/v2/listener_test.go:2]: 2024

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**